src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -237,9 +240,44 @@ def create_keyvault(client,
                                         KeyPermissions.update,
                                         KeyPermissions.import_enum,
                                         KeyPermissions.backup,
-                                        KeyPermissions.restore],
-                                  secrets=[SecretPermissions.all],
-                                  certificates=[CertificatePermissions.all])
+                                        KeyPermissions.restore,
+                                        KeyPermissions.recover,
+                                        KeyPermissions.purge],
+                                  secrets=[
+                                      SecretPermissions.get,
+                                      SecretPermissions.list,
+                                      SecretPermissions.set,
+                                      SecretPermissions.delete,
+                                      SecretPermissions.backup,
+                                      SecretPermissions.restore,
+                                      SecretPermissions.recover,
+                                      SecretPermissions.purge],
+                                  certificates=[
+                                      CertificatePermissions.get,
+                                      CertificatePermissions.list,
+                                      CertificatePermissions.delete,
+                                      CertificatePermissions.create,
+                                      CertificatePermissions.import_enum,
+                                      CertificatePermissions.update,
+                                      CertificatePermissions.managecontacts,
+                                      CertificatePermissions.getissuers,
+                                      CertificatePermissions.listissuers,
+                                      CertificatePermissions.setissuers,
+                                      CertificatePermissions.deleteissuers,
+                                      CertificatePermissions.manageissuers,
+                                      CertificatePermissions.recover,
+                                      CertificatePermissions.purge],
+                                  storage=[
+                                      StoragePermissions.get,
+                                      StoragePermissions.list,
+                                      StoragePermissions.delete,
+                                      StoragePermissions.set,
+                                      StoragePermissions.update,
+                                      StoragePermissions.regeneratekey,
+                                      StoragePermissions.setsas,
+                                      StoragePermissions.listsas,
+                                      StoragePermissions.getsas,
+                                      StoragePermissions.deletesas])
```

Comment:
All the permissions enumerations used here are defined in the SDK, we explicitly list out the values because the permissions granted by default are not all available permissions.